### DISCUSSION TITLE
What are the top 10 highest priority vLEI related KERI issues? - Updated 20240909

### DISCUSSION THREAD

**Discussion by 2byrds** [WebOfTrust/keri#78] [Ideas]
#### UPDATED Priority list:
1. High: Integration Tests are failing on 0.4.0 of KERIA: https://github.com/WebOfTrust/signify-ts/issues/392
1. High, Production: https://github.com/WebOfTrust/keria/issues/233
1. High, Production: https://github.com/WebOfTrust/keria/issues/175
1. High, Partially implemented: https://github.com/WebOfTrust/keria/issues/37
1. High: https://github.com/WebOfTrust/keria/issues/232
1. High: https://github.com/WebOfTrust/keripy/issues/763
1. Medium: https://github.com/WebOfTrust/keria/issues/217

This is now tracked as a project here https://github.com/orgs/GLEIF-IT/projects/10/views/2

~~Please respond here with any issue (no matter how old) from a KERI-related repo that you would like highlighted at the KERI dev meeting on Thursdays. To gamify it we will highlight the issue at the meeting and then discuss if it belongs in the top 10 vLEI related issues.~~

IN REVIEW:
1. Blocker, Production, Intermittent: https://github.com/WebOfTrust/keria/issues/231
1. Blocker, Production: https://github.com/WebOfTrust/keria/issues/222

Needs Consideration:
1. Rollback of partially signed events: https://github.com/WebOfTrust/keria/issues/213 - to be able to correct user mistakes, this is crucial. Otherwise we cannot recover a multisig AID when two of their members have diverging KELs for the the group.
1. Filter operation by status: https://github.com/WebOfTrust/keria/issues/192 - to be able to inform the users that there are unresolved operations. This can be done by filtering on the client side, but it will not work forever as long as the list of operations grows.
1. https://github.com/WebOfTrust/keria/issues/240
1. https://github.com/WebOfTrust/keria/issues/283

Previously Considered:
1. Medium, Production, Partially addressed: https://github.com/WebOfTrust/signify-ts/issues/204
1. Medium, Workaround available: https://github.com/WebOfTrust/keria/issues/171
1. Medium: https://github.com/WebOfTrust/keria/issues/200

Thank you for previous feedback from @lenkan @kentbull @nkongsuwan @pfeairheller @daviddm

FIXED:
- https://github.com/WebOfTrust/keripy/issues/773
- Blocker, Production, Intermittent: https://github.com/WebOfTrust/keripy/issues/762
- https://github.com/WebOfTrust/keria/issues/201
- https://github.com/WebOfTrust/keripy/issues/694
- https://github.com/WebOfTrust/keripy/issues/734
- https://github.com/WebOfTrust/signify-browser-extension/issues/155
- Blocker, Depends on 694 which is fixed: https://github.com/WebOfTrust/keria/issues/230
- Blocker, Production: https://github.com/WebOfTrust/signify-ts/issues/227
- Blocker, Production: https://github.com/WebOfTrust/keria/issues/257
- Medium: https://github.com/WebOfTrust/keria/issues/184
- Blocker, Production: https://github.com/WebOfTrust/keria/issues/235

**Comment by kentbull** [comment]
Nice work lance! This list is really shaping up.

**Comment by 2byrds** [comment]
Possibly add labels to issues to add them for consideration. For instance we use [effort-eval-needed](https://github.com/WebOfTrust/keria/labels/effort-eval-needed) now.
An if you have an idea on the LOE, etc.

**Comment by lenkan** [comment]
Also added this https://github.com/WebOfTrust/keria/issues/235 as a follow up from https://github.com/WebOfTrust/keripy/issues/762

**Comment by edeykholt** [comment]
Should we list strategic priorities as a separate list?   Maybe not stack ranked, and what would help the ecosystem more generally.

- KERIA hot standby /  https://github.com/WebOfTrust/keria/issues/240
- Watcher Network
- ...

**Comment by lenkan** [comment]
During testing, we have found two issues that we think needs priority:

Rollback of partially signed events: https://github.com/WebOfTrust/keria/issues/213 - to be able to correct user mistakes, this is crucial. Otherwise we cannot recover a multisig AID when two of their members have diverging KELs for the the group.

Filter operation by status: https://github.com/WebOfTrust/keria/issues/192 - to be able to inform the users that there are unresolved operations. This can be done by filtering on the client side, but it will not work forever as long as the list of operations grows.

**Reply by 2byrds** [reply]
> During testing, we have found two issues that we think needs priority:
> 
> Rollback of partially signed events: [WebOfTrust/keria#213](https://github.com/WebOfTrust/keria/issues/213) - to be able to correct user mistakes, this is crucial. Otherwise we cannot recover a multisig AID when two of their members have diverging KELs for the the group.
> 
> Filter operation by status: [WebOfTrust/keria#192](https://github.com/WebOfTrust/keria/issues/192) - to be able to inform the users that there are unresolved operations. This can be done by filtering on the client side, but it will not work forever as long as the list of operations grows.

Thank you @lenkan I have added those two to the 'Needs Consideration' list

**Comment by lenkan** [comment]
@2byrds and others. What do you think about converting this into a github project? I would like to track issues beyond their closures, and also be able to have a better overview of what is planned for the future and how that work is being prioritized.

At the moment, when an issue is closed because a PR is merged, it does not always mean that the work is completed. Often, it needs further reviewing, patching of old release branch, update of downstream repositories, and last but not least a lot of testing. In a github project, you can represent these stages as different columns.

As for the priority and backlog - at the moment there are a lot of older issues across the repos that make up the vLEI ecosystem. Most of these do not receive any attention. Some of them can be closed, some of them might be easy fixes.

**Reply by 2byrds** [reply]
> @2byrds and others. What do you think about converting this into a github project? I would like to track issues beyond their closures, and also be able to have a better overview of what is planned for the future and how that work is being prioritized.
> 
> At the moment, when an issue is closed because a PR is merged, it does not always mean that the work is completed. Often, it needs further reviewing, patching of old release branch, update of downstream repositories, and last but not least a lot of testing. In a github project, you can represent these stages as different columns.
> 
> As for the priority and backlog - at the moment there are a lot of older issues across the repos that make up the vLEI ecosystem. Most of these do not receive any attention. Some of them can be closed, some of them might be easy fixes.

@lenkan I support that idea. I have tracked the priority, etc. of several of these issues in a github project related to our joint work but doing it as a vLEI community project makes sense to me.

**Reply by 2byrds** [reply]
Okay the project is here @lenkan  https://github.com/orgs/GLEIF-IT/projects/10/views/2
![image](https://github.com/user-attachments/assets/294c1839-09c0-4fde-9783-da2073e9896c)
![image](https://github.com/user-attachments/assets/97ae5db6-31b5-42ae-86f2-575a9a4fc495)
